src/classes/BDI_DataImportService.cls

```diff
@@ -704,17 +704,40 @@ global with sharing class BDI_DataImportService {
     public static Boolean failFieldLevelSecurity = false;
 
     /*******************************************************************************************************
-    * @description returns the target object (bdi name) for a given Data Import field
+    * @description returns the target object (api name) for a given Data Import field
     * @param sourceFieldName the api name of the Data Import field
-    * @return String the BDI name of the target object for that field, derived from the help text
+    * @return String the api name of the target object for that field, derived from the help text
     */
     public static String getTargetObject(String sourceFieldName) {
         sourceFieldName = sourceFieldName.toLowerCase();
         String dataImportObjectName = UTIL_Namespace.StrTokenNSPrefix('DataImport__c');
         DescribeFieldResult dfr = UTIL_Describe.getFieldDescribe(dataImportObjectName, sourceFieldName);
         String targetObject = dfr.inlineHelpText != null ? dfr.inlineHelpText.split('\\.')[0] : null;
 
-        return targetObject;
+        // Donation Campaign is a special case because its mapping is hard-coded, not in help text
```

Comment:
What's the reason we aren't using a help text mapping like other fields?

Reply:
Because it needs to work both as an input from BGE and as an output from BDI if they use the legacy campaign name matching.

Reply:
Even legacy records should work using the help text mapping - as soon as they run matching (through dry run or import) existing records should be updated to populate the Id fields. /cc @randi274 for confirmation

Reply:
Can we discuss this on a call? I'm not understanding the question. The campaign name-matching feature doesn't use help text mapping either.

Reply:
Removed, as we've adjusted things to rely on help text for Campaign ID mapping.